### PR TITLE
[FLINK-30232][Build/Shaded] Include all platforms incl ARM64 while shading Netty

### PR DESCRIPTION
**Pull request by MartijnVisser**

### CODE REVIEW

File: flink-shaded-netty-4/pom.xml [hunk 1]
```diff
@@ -94,12 +94,12 @@ under the License.
...
-                                    <mapper type="glob" from="libnetty_transport_native_epoll_x86_64.so" to="liborg_apache_flink_shaded_netty4_netty_transport_native_epoll_x86_64.so"/>
+                                    <mapper type="glob" from="libnetty_transport_native_epoll_*.so" to="liborg_apache_flink_shaded_netty4_netty_transport_native_epoll_*.so"/>
                                 </move>
                                 <echo message="renaming native kqueue library" />
                                 <move todir="${project.build.directory}/unpacked/META-INF/native" includeemptydirs="false">
                                     <fileset dir="${project.build.directory}/unpacked/META-INF/native"/>
-                                    <mapper type="glob" from="libnetty_transport_native_kqueue_x86_64.jnilib" to="liborg_apache_flink_shaded_netty4_netty_transport_native_kqueue_x86_64.jnilib"/>
+                                    <mapper type="glob" from="libnetty_transport_native_kqueue_*.jnilib" to="liborg_apache_flink_shaded_netty4_netty_transport_native_kqueue_*.jnilib"/>
```
Comment (zentol) [inline flink-shaded-netty-4/pom.xml:102]: if these patterns actually work we could merge the epoll/kqueue moves and significantly streamline things.

```
from:  libnetty*
to:    liborg_apache_flink_shaded_netty*
```

Comment (zentol) [reply flink-shaded-netty-4/pom.xml:102]: Not sure if there is other stuff in the native directory that we wouldnt want relocated.

**Review by zentol — CHANGES_REQUESTED** [review]
This doesn't cover all native files of netty4. There are some macos-specific resolved libraries that also need to be relocated.
If you follow my suggestion these should be automatically covered.

The same streamlining should be applied to the dynamic tcnative module, whereas the static tcnative module is also missing a relocation for macos.

**Comment by MartijnVisser** [comment]
> The same streamlining should be applied to the dynamic tcnative module

We can't do that for the dynamic tcnative module, per https://github.com/apache/flink-shaded/blob/master/flink-shaded-netty-tcnative-dynamic/pom.xml#L157 - I've tried to streamline it but then it doesn't work properly. The other comments have been taken care off

**Review by zentol — APPROVED** [review]

**Pull request merged**